projects/plugins/jetpack/modules/search/instant-search/components/dom-event-handler.jsx

```diff
@@ -0,0 +1,199 @@
+		} );
+	}
+
+	removeEventListeners() {
+		window.removeEventListener( 'popstate', this.handleHistoryNavigation );
+
+		document.querySelectorAll( this.props.themeOptions.searchInputSelector ).forEach( input => {
+			input.form.removeEventListener( 'submit', this.handleSubmit );
+			input.removeEventListener( 'keyup', this.handleKeyup );
+			input.removeEventListener( 'input', this.handleInput );
+			input.removeEventListener( 'compositionstart', this.handleCompositionStart );
+			input.removeEventListener( 'compositionend', this.handleCompositionEnd );
+		} );
+
+		document.querySelectorAll( this.props.themeOptions.overlayTriggerSelector ).forEach( button => {
+			button.removeEventListener( 'click', this.handleOverlayTriggerClick, true );
+		} );
+
+		document.querySelectorAll( this.props.themeOptions.filterInputSelector ).forEach( element => {
+			element.removeEventListener( 'click', this.handleFilterInputClick );
+		} );
+
+		document.querySelectorAll( `.${ OVERLAY_CLASS_NAME }` ).forEach( element => {
+			element.removeEventListener( 'transitionend', this.scrollOverlayToTop );
+		} );
+	}
+
+	scrollOverlayToTop( event ) {
+		if ( event?.propertyName !== 'transform' ) {
+			return;
+		}
+		const overlay = event.target;
+		// NOTE: IE11 doesn't support scrollTo. Manually set overlay element's scrollTop.
+		if ( overlay.scrollTo ) {
+			// @see https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollTo
+			overlay.scrollTo( 0, 0 );
+		} else {
+			overlay.scrollTop = 0;
+		}
+	}
+
+	handleCompositionStart = () => this.setState( { isComposing: true } );
+	handleCompositionEnd = () => this.setState( { isComposing: false } );
+
+	handleFilterInputClick = event => {
+		event.preventDefault();
+		if ( event.currentTarget.dataset.filterType ) {
+			if ( event.currentTarget.dataset.filterType === 'taxonomy' ) {
+				this.props.setFilter(
+					event.currentTarget.dataset.taxonomy,
+					event.currentTarget.dataset.val
+				);
+			} else {
+				this.props.setFilter(
+					event.currentTarget.dataset.filterType,
+					event.currentTarget.dataset.val
+				);
+			}
+		}
+		this.showResults();
+	};
+
+	handleHistoryNavigation = () => {
+		// Treat history navigation as brand new query values; re-initialize.
+		// Note that this re-initialization will trigger onChangeQueryString via side effects.
+		this.props.initializeQueryValues( { isHistoryNavigation: true } );
+	};
+
+	handleInput = debounce( event => {
+		// Reference: https://rawgit.com/w3c/input-events/v1/index.html#interface-InputEvent-Attributes
+		// NOTE: inputType is not compatible with IE11, so we use optional chaining here. https://caniuse.com/mdn-api_inputevent_inputtype
+		if ( event.inputType?.includes( 'format' ) || event.target.value === '' ) {
+			return;
+		}
+
+		// Is the user still composing input with a CJK language?
+		if ( this.state.isComposing ) {
+			return;
+		}
+
+		if ( this.state.overlayOptions.overlayTrigger === 'submit' ) {
```

Comment:
The line triggers the following error in console and overlay wouldn't popup.
![undefinded-overlay-options](https://user-images.githubusercontent.com/1425433/121109400-990ac400-c85f-11eb-9c71-3ebd3c2c4501.jpg)

`overlayOptions` doesn't seem to be defined as a state, probably need to be passed as a props from search-app.

Reply:
Good catch! I discovered and fixed several more scope issues with DomEventHandler while fixing this.